<launch>
  <!--
    Push the left frontal button labeled as 'LB' to activate cmd_vel publishing.
    Move the left stick around to control the velocity.
   -->

  <node pkg="myrobot_teleop" type="myrobot_teleop_joy" name="myrobot_teleop_joystick">

    <param name="axis_left_lr_joy" value="0"/>
    <param name="axis_left_ud_joy" value="1"/>
    <param name="axis_right_lr_joy" value="2"/>
    <param name="axis_right_ud_joy" value="3"/>
    <param name="axis_right_throttle" value="4"/>
    <param name="axis_left_throttle" value="5"/>
    
    <param name="button_a" value="0"/>
	<param name="button_b" value="1"/>    
    <param name="button_x" value="2"/>
    <param name="button_y" value="3"/>

    <param name="max_speed" value="0.5"/>
    <param name="max_omega" value="0.5"/>
    <remap from="myrobot_teleop_joystick/cmd_vel" to="cmd_vel/joy_teleop"/>
    <remap from="myrobot_teleop_joystick/move_base/cancel" to="move_base/cancel"/>
    <remap from="myrobot_teleop_joystick/move_base/goal" to="move_base/goal"/>
    
  </node>

  <node pkg="joy" type="joy_node" name="joystick">
    <param name="dev" value="/dev/input/js0"/>
    <param name="autorepeat_rate" value="5.0"/>
  </node>



</launch>
